### DISCUSSION TITLE
705

### DISCUSSION THREAD

**Discussion by giscus** [jojoldu/blog-comments#887] [Comments]
# 705

최근에 Giscus로 댓글 시스템을 이관하면서 새 댓글이 생성될때 알람을 받을 방법이 없었다. 그래서 간단하게 Github Action과 Telegram을 통해 알람을 받도록 구성했다. 1. 텔레그램 Bot 생성 BotFather 에서 새로운 봇을 생성한다. /newbot 을 입력해서 새로운 봇 생성을 시작 blog-comment 라는 이름으로 봇 생성 방금전에 생성한 blog-comment 봇의 username을 지어야한다. 여기서는 MyBlogCommentBot 으로 지었다. 마지막엔 꼭 bot이 붙어야 한다. 마지막 메세지에 포함된 token 값이 앞으로 사용할 Token이라 별도로 관리를 한다. 해당 Bot의 채널ID를 알아야 API를 사용할 수 있다. chatId를 확인해보기 위해 생성된 Bot..

https://jojoldu.tistory.com/705

**Comment by kdohyeon** [comment]
좋은 글 잘 읽었습니다! (댓글 알람)

**Comment by ShanePark** [comment]
감사합니다. 덕분에 잘 참고해서 Slack에 알람보내는 액션 생성했습니다. 기본 슬랙 깃헙봇 만으로는 댓글은 안되고 새로운 Discussions이 생성되는 `if: ${{ github.event.discussion && !github.event.comment }}` 것만 알람이 되더라고요. 

한가지, 댓글 알림 쪽에 수정이 필요해 보여요 

**변경전**
 ```yaml
USER_URL: ${{ github.event.discussion.user.html_url }}
USER_NAME: ${{ github.event.discussion.user.login }}
```
**변경 후**
```yaml
USER_URL: ${{ github.event.comment.user.html_url }}
USER_NAME: ${{ github.event.comment.user.login }}
```
지금 상태에서는 댓글 알람시에 댓글 작성자의 아이디/프로필이 아닌 Discussion 작성자 `giscus[bot]` 만 계속 나올 것 같아요

**Reply by jojoldu** [reply]
아고! 확인 감사합니다 :) 본문 코드 바로 수정하겠습니다! @ShanePark 감사합니다 :)

**Comment by umbum** [comment]
공유 감사합니다. 많은 도움이 되었습니다.

**Comment by kimgunwooo** [comment]
좋은 글 잘 읽었습니다~
